common/dynamicconfig/memory_client.go

```diff
@@ -29,32 +29,40 @@ import (
 )
 
 type MemoryClient struct {
-	sync.RWMutex
+	lock      sync.RWMutex
 	overrides map[Key]any
 	fallback  Client
 }
 
 func (d *MemoryClient) GetRawValue(name Key) (any, bool) {
-	d.RLock()
-	defer d.RUnlock()
+	d.lock.RLock()
+	defer d.lock.RUnlock()
 	v, ok := d.overrides[name]
+	if value, ok := v.([]ConstrainedValue); ok {
+		return value[0].Value, true
```

Comment:
this seems wrong, you can't just pick the first one.. also this whole function is unused now?

oh, I see where this is used. uugh, that's messy. I think you can just use GetValue there and get rid of the "must override" check. pick the first value from the ConstrainedValues with len(Constraints)==0, and panic if there is none.

(the idea was that those tests always should be explicit about partition count since the behavior can be pretty different with 1 vs many partitions, and I wanted most tests to use 1 partition. but I think it doesn't really matter that much.)

Reply:
Done